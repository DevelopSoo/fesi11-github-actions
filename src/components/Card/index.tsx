import clsx from "clsx";

type CardProps = {
  children: React.ReactNode;
  variant?: "default" | "outlined" | "elevated";
  padding?: "none" | "sm" | "md" | "lg";
  radius?: "none" | "sm" | "md" | "lg" | "full";
  className?: string;
};

export default function Card({
  children,
  variant = "default",
  padding = "md",
  radius = "md",
  className,
}: CardProps) {
  const cardClasses = clsx(
    "overflow-hidden transition-all bg-white",
    {
      "p-0": padding === "none",
      "p-3": padding === "sm",
      "p-5": padding === "md",
      "p-8": padding === "lg",
    },
    {
      "rounded-none": radius === "none",
      "rounded-sm": radius === "sm",
      "rounded-md": radius === "md",
      "rounded-lg": radius === "lg",
      "rounded-full": radius === "full",
    },
    {
      "border border-gray-200": variant === "default",
      "border border-gray-500 hover:border-gray-400 transition-colors":
        variant === "outlined",
      "shadow-lg hover:shadow-xl transition-shadow": variant === "elevated",
    },
    className,
  );
  return <div className={cardClasses}>{children}</div>;
}
