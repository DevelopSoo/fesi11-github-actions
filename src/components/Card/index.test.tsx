import { render } from "@testing-library/react";
import Card from ".";

test("카드 스냅샷", () => {
  const cardProps = {
    title: "제목",
    description: "설명",
    imageUrl: "https://example.com/image.jpg",
  };
  const { container } = render(<Card {...cardProps} />);
  expect(container).toMatchSnapshot();
});
