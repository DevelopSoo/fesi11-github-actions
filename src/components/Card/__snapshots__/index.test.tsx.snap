// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`카드 스냅샷 1`] = `
<div>
  <div
    class="rounded-lg bg-white p-4 shadow-md"
  >
    <h2
      class="text-lg font-bold"
    >
      제목
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      설명
    </p>
    <img
      alt="제목"
      class="h-40 w-full object-cover"
      src="https://example.com/image.jpg"
    />
  </div>
</div>
`;
